### PR TITLE
Input Feature Noise Augmentation — Gaussian Noise on All Input Channels

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The model processes ~26 input features per mesh node (positions, DSDF channels, curvature, distance-to-surface, etc.). The val_tandem_transfer test evaluates on NACA6416 — an unseen foil shape that produces slightly different feature distributions than the training shapes. Even small distribution shifts in input features can cause disproportionate prediction errors in a well-optimized model.

**Input noise augmentation** adds small Gaussian noise to ALL input features AFTER standardization during training. This is a standard regularization technique (similar to how image models use pixel noise) that:
1. Prevents overfitting to the exact feature values in the training set
2. Smooths the loss landscape in feature space → better generalization
3. Makes the model robust to small distribution shifts in any feature → directly targets the OOD failure mode

This is fundamentally different from DSDF-specific augmentations (which target geometry features only) — it regularizes the ENTIRE input representation. It's also extremely cheap (~2 LoC, zero computational overhead).

**Expected impact:** p_tan -0.5% to -2%. Strongest effect expected on OOD metrics (p_oodc, p_re, p_tan) where input distribution shift is the bottleneck.

### Step 1: Add config flag

```python
# Config dataclass:
input_noise_sigma: float = 0.0  # std of Gaussian noise added to standardized input features
```

```python
# argparse:
parser.add_argument('--input_noise_sigma', type=float, default=0.0,
                    help='Gaussian noise std added to standardized input features during training')
```

### Step 2: Apply noise AFTER standardization

Find the standardization line: `x = (x - stats["x_mean"]) / stats["x_std"]` (~line 1568).

IMMEDIATELY AFTER it, add:

```python
if cfg.input_noise_sigma > 0.0 and model.training:
    x = x + torch.randn_like(x) * cfg.input_noise_sigma
```

That's it. 2 lines of code.

**Important:**
- Apply AFTER standardization — noise magnitude is relative to the standardized features (unit-ish scale)
- The `model.training` guard ensures NO noise during validation/evaluation
- σ=0.01 means ~1% perturbation of each standardized feature per node per sample

### Step 3: Run 6 experiments (3 noise levels × 2 seeds)

Sweep: σ = {0.01, 0.03, 0.05}

```bash
# σ=0.01, seed 42
cd cfd_tandemfoil && python train.py --agent edward \
  --wandb_name "edward/innoise01-s42" --wandb_group phase6/input-noise-aug \
  --input_noise_sigma 0.01 --seed 42 \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15

# σ=0.01, seed 73 (same, --seed 73, wandb_name "edward/innoise01-s73")

# σ=0.03, seed 42 (same, --input_noise_sigma 0.03, wandb_name "edward/innoise03-s42")
# σ=0.03, seed 73 (same, --seed 73, wandb_name "edward/innoise03-s73")

# σ=0.05, seed 42 (same, --input_noise_sigma 0.05, wandb_name "edward/innoise05-s42")
# σ=0.05, seed 73 (same, --seed 73, wandb_name "edward/innoise05-s73")
```

**Notes:**
- Do NOT include `--aft_foil_srf_context` (confirmed harmful, removed from baseline)
- DO include `--pcgrad_3way --pcgrad_extreme_pct 0.15` and `--aug_dsdf2_sigma 0.05` (current baseline)
- If limited to 4 GPUs, prioritize σ=0.01 and σ=0.03 (2 seeds each), add σ=0.05 if time permits.

Report: 2-seed avg for each σ value across all 4 metrics + W&B run IDs.

## Baseline

Current single-model baseline (PR #2119, PCGrad 2-way):

| Metric | 2-seed target |
|--------|--------------|
| p_in   | < 13.20      |
| p_oodc | < 7.91       |
| p_tan  | **< 29.48**  |
| p_re   | < 6.50       |

Baseline W&B: jpe1t13t (s42), cdccuyl7 (s73)

```bash
cd cfd_tandemfoil && python train.py --agent <name> --wandb_name "<name>/baseline" \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15
```

### CODE REVIEW

**Comment by tcapelle** [comment]
## Experiment Results: Input Feature Noise Augmentation

### Setup
- 6 runs: 3 noise levels (σ=0.01, 0.03, 0.05) × 2 seeds (42, 73)
- All runs: `--ema_start_epoch 100 --ema_decay 0.999 --cosine_T_max 160 --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 --pcgrad_3way --pcgrad_extreme_pct 0.15` + standard baseline flags
- Noise applied: `x = x + randn_like(x) * sigma` immediately after standardization, training only
- All runs reached epoch 145-147 before 180-min timeout

### Results — Surface MAE (Pressure)

| Run | p_in | p_oodc | p_tan | p_re | val/loss | W&B Run ID |
|-----|------|--------|-------|------|----------|------------|
| σ=0.01 s42 | 15.8 | 9.6 | 31.2 | 8.0 | 0.430 | `ibpmve5e` |
| σ=0.01 s73 | 15.2 | 9.6 | 30.2 | 7.9 | 0.426 | `cxyu1uvk` |
| **σ=0.01 avg** | **15.5** | **9.6** | **30.7** | **7.95** | — | — |
| σ=0.03 s42 | 26.0 | 11.0 | 32.3 | 9.3 | 0.500 | `0hfbsp66` |
| σ=0.03 s73 | 27.9 | 10.9 | 32.3 | 8.9 | 0.508 | `o3embbbn` |
| **σ=0.03 avg** | **26.95** | **10.95** | **32.3** | **9.1** | — | — |
| σ=0.05 s42 | 22.7 | 11.8 | 31.6 | 9.8 | 0.501 | `d2otilbo` |
| σ=0.05 s73 | 24.3 | 11.6 | 33.0 | 9.5 | 0.511 | `fgczv5yq` |
| **σ=0.05 avg** | **23.5** | **11.7** | **32.3** | **9.65** | — | — |
| **Baseline** | **13.20** | **7.91** | **29.48** | **6.50** | — | — |

### Analysis

**Strong negative result.** All noise levels cause significant degradation across all metrics:

| Metric | σ=0.01 | σ=0.03 | σ=0.05 | Baseline |
|--------|--------|--------|--------|----------|
| p_in | +17.4% | +104% | +78% | 13.20 |
| p_oodc | +21.4% | +38.4% | +47.9% | 7.91 |
| p_tan | +4.1% | +9.6% | +9.6% | 29.48 |
| p_re | +22.3% | +40.0% | +48.5% | 6.50 |

Even σ=0.01 (1% perturbation of standardized features) causes a 17% regression on p_in and 22% on p_re.

**Why this fails:**
1. **CFD inputs are NOT like image pixels.** In image classification, pixel noise is bounded and the model needs to learn invariance to it. In CFD mesh prediction, the input features (positions, signed distance fields, curvature) carry precise geometric information that the model must faithfully map to flow solutions. Adding noise corrupts the geometry→flow mapping.
2. **DSDF channels are especially sensitive.** The signed distance field encodes exact distances to foil surfaces. Noise on these features effectively tells the model that nodes near the surface are further away (or vice versa), creating inconsistent geometry→flow pairs that hurt learning.
3. **The model already generalizes via augmentation.** The existing `aoa_perturb`, `aug_full_dsdf_rot`, `aug_gap_stagger_sigma`, and `aug_dsdf2_sigma` augmentations already provide physically-consistent regularization. Generic input noise is inferior because it's physically inconsistent.
4. **Monotonic degradation.** The damage increases with σ (except σ=0.05 on p_in, likely due to under-fitting dominating over-fitting). This confirms the noise is pure corruption, not regularization.

### Conclusion

**Close this direction.** Input feature noise augmentation is fundamentally incompatible with this problem domain. The model needs precise geometric features to predict flow fields. Any future augmentation should be physically consistent (e.g., actual geometry perturbations with re-simulated targets).

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2144 because ADVISOR: Closing — input feature noise is catastrophically harmful. Even σ=0.01 causes p_in +17%, p_re +22%.

Your analysis is excellent: CFD mesh features carry precise geometry that maps deterministically to flow fields. Adding noise creates inconsistent geometry→flow pairs that degrade learning — fundamentally different from image noise augmentation. The existing augmentations (aoa_perturb, dsdf_rot, gap_stagger_sigma, dsdf2_sigma) all work BECAUSE they're physically consistent. Generic noise is not.

**Dead end confirmed.** This also rules out any form of generic input perturbation (dropout on features, noise injection, etc.) — we already confirmed DSDF spatial dropout (#2143) fails for the same reason.

New assignment coming shortly — you'll be testing a learning rate sweep..
